### ISSUE TITLE
interpolate/sample array at point

### ISSUE THREAD

**Issue by cossatot** [pydata/xarray#191]
This looks like an excellent project, thanks for all the work.

I am wondering if there is functionality for interpolation or sampling of a Dataset at a point (or sequence of points).  The SciPy way of doing this uses [scipy.ndimage.map_coordinates](http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.interpolation.map_coordinates.html) and is completely backwards for (geo)spatial data, requiring the point's coordinates to be converted to 'array' coordinates before interpolation.

Does this already have an implementation, given how super common an operation it is?  If not, would it be difficult to wrap [pandas.Series.interpolate()](http://pandas.pydata.org/pandas-docs/dev/generated/pandas.Series.interpolate.html) up?  I might write a function to do this, but I don't known the xray internals well enough to know where to stick it.

Thanks,
Richard

**Comment by denfromufa** [comment]
+1

**Comment by saulomeirelles** [comment]
Hi All,

This is indeed an excellent project with great potential! 

I am wondering if there is any progress on the interpolation issue. I am working with an irregular time series which I would pretty much like to upsample it using xray.

Thanks for all the effort!

Saulo

**Comment by shoyer** [comment]
@saulomeirelles I don't have any progress to share on this issue. If this issue is important to you, contributions would be gratefully accepted!

**Comment by sjpfenninger** [comment]
I've written a wrapper around scipy's ``map_coordinates``. I cleaned this up a bit and pushed it here: https://github.com/sjpfenninger/xray/commit/0e0d88d6dc6f428d24ae3df2161ff10deded2a5a

It's not exactly fully-featured but scratches the itch I had, which is to pass DataArrays through map_coordinates with support for xarray's coordinates.

@shoyer If this seems of use, I could add some tests and perhaps an example, then submit a pull request?

**Comment by shoyer** [comment]
Yes, this like looks useful to me!

**Comment by sjpfenninger** [comment]
Ok, I'll extend this into a pull request one of these days

**Comment by jgerardsimcock** [comment]
I have a dataset that looks like the following:

```
<xarray.Dataset>
Dimensions:  (lat: 720, lon: 1440, time: 365)
Coordinates:
  * time     (time) datetime64[ns] 2006-01-01T12:00:00 2006-01-02T12:00:00 ...
  * lat      (lat) float32 -89.875 -89.625 -89.375 -89.125 -88.875 -88.625 ...
  * lon      (lon) float32 0.125 0.375 0.625 0.875 1.125 1.375 1.625 1.875 ...
Data variables:
    tasmax   (time, lat, lon) float64 272.6 272.6 272.6 272.6 272.6 272.6 ...
Attributes:
    parent_experiment:              historical
    parent_experiment_id:           historical
    parent_experiment_rip:          r1i1p1
    Conventions:                    CF-1.4
    institution:                    NASA Earth Exchange, NASA Ames Research C...
    institute_id:                   NASA-Ames
    realm:                          atmos
    modeling_realm:                 atmos
    version:                        1.0
    downscalingModel:               BCSD
    experiment_id:                  rcp85
    frequency:                      day
    realization:                    1
    initialization_method:          1
    physics_version:                1
    tracking_id:                    fd966a07-baec-44e4-8f69-e3cfb2d70dfa
    driving_data_tracking_ids:      N/A
    driving_model_ensemble_member:  r1i1p1
    driving_experiment_name:        historical
    driving_experiment:             historical
    model_id:                       BCSD
    references:                     BCSD method: Thrasher et al., 2012, Hydro...
    DOI:                            http://dx.doi.org/10.7292/W0MW2F2G
    experiment:                     RCP8.5
    title:                          ACCESS1-0 global downscaled NEX CMIP5 Cli...
    contact:                        Dr. Rama Nemani: rama.nemani@nasa.gov, Dr...
    disclaimer:                     This data is considered provisional and s...
    resolution_id:                  0.25 degree
    project_id:                     NEXGDDP
    table_id:                       Table day (12 November 2010)
    source:                         BCSD 2014
    creation_date:                  2015-01-07T20:16:06Z
    forcing:                        N/A
    product:                        output
```

I am trying to do a linear interpolation for each day where the temp is nan. Is there a straightforward way to do this in Xarray?

**Comment by max-sixty** [comment]
@fujiisoup could we close this given your recent work

**Comment by fujiisoup** [comment]
closed via #2104
